flows/src/test/java/com/softwaremill/jox/FlowsTest.java

```diff
@@ -0,0 +1,226 @@
+package com.softwaremill.jox;
+
+import com.softwaremill.jox.structured.Fork;
+import com.softwaremill.jox.structured.UnsupervisedScope;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class FlowsTest {
+
+    @Test
+    void shouldBeEmpty() throws Exception {
+        assertTrue(Flows.empty().runToList().isEmpty());
+    }
+
+    @Test
+    @WithUnsupervisedScope
```

Comment:
does this do anything other than calling `Scopes.unsupervised(...)`? If so, it only saves one line, so maybe not worth it?

Reply:
Yes, it only calls Scopes.unsupervised and passes scope as argument.

IMO it makes the tests a little bit cleaner, there is no need to wrap the whole test in lambda and there is no ugly return null at the end. 

So 
```java
    @Test
    void shouldCreateFlowFromFork() throws Exception {
        Scopes.unsupervised(scope -> {
            Fork<Integer> f = scope.forkUnsupervised(() -> 1);
            Flow<Integer> c = Flows.fromFork(f);
            assertEquals(List.of(2), c.runToList());
            return null;
        });
    }
``` 
vs 

```java
    @Test
    @WithUnsupervisedScope
    void shouldCreateFlowFromFork(UnsupervisedScope scope) throws Exception {
        Fork<Integer> f = scope.forkUnsupervised(() -> 1);
        Flow<Integer> c = Flows.fromFork(f);
        assertEquals(List.of(1), c.runToList());
    }
```

Reply:
Yeah the `return null` is indeed ugly, but I think I'd opt for explicit `Scopes.unsupervised` calls, as that's how the flows/scopes should be used. That is, the scope is not some context that "normally" will be available from another place, but rather the primary means of usage.

Reply:
Ok, makes sense, will change it then

Reply:
done